pkg/library/overrides/pods.go

```diff
@@ -72,55 +73,68 @@ func ApplyPodOverrides(workspace *common.DevWorkspaceWithConfig, deployment *app
 	patched.Spec.Template = patchedPodSpecTemplate
 	patched.Spec.Template.Spec.Containers = originalContainers
 	patched.Spec.Template.Spec.InitContainers = originalInitContainers
-	patched.Spec.Template.Spec.Volumes = originalVolumes
 	return patched, nil
 }
 
+func GetPodVolumeOverrides(workspace *dw.DevWorkspaceTemplateSpec) (*[]corev1.Volume, error) {
```

Comment:
We probably don't want to return a reference to a slice here (`*[]corev1.Volume`), `[]corev1.Volume` should be enough (since a slice is already a reference).